dev/ssl_darwin.go

```diff
@@ -1,62 +1,69 @@
 package dev
 
 import (
-	"errors"
+	"bytes"
 	"fmt"
-	"io/ioutil"
-	"os"
 	"os/exec"
-
-	"github.com/puma/puma-dev/homedir"
 )
 
+// SupportDir is the platform-specific path that contains puma-dev's generated certs
 const SupportDir = "~/Library/Application Support/io.puma.dev"
 
-func LoginKeyChain() (string, error) {
-
-	new_keychain_path := homedir.MustExpand("~/Library/Keychains/login.keychain-db")
-	old_keychain_path := homedir.MustExpand("~/Library/Keychains/login.keychain")
-
-	if _, err := os.Stat(new_keychain_path); err == nil {
-		return new_keychain_path, nil
-	}
-
-	if _, err := os.Stat(old_keychain_path); err == nil {
-		return old_keychain_path, nil
-	}
-
-	return "", errors.New("Could not find login keychain")
-}
-
+// TrustCert adds the cert at the provided path to the macOS default login keychain
 func TrustCert(cert string) error {
 	fmt.Printf("* Adding certification to login keychain as trusted\n")
 	fmt.Printf("! There is probably a dialog open that requires you to authenticate\n")
 
-	login, keychainError := LoginKeyChain()
+	login, keychainError := loginKeyChain()
 
 	if keychainError != nil {
 		return keychainError
 	}
 
-	addTrustedCertCommand := exec.Command("sh", "-c",
-		fmt.Sprintf(`security add-trusted-cert -d -r trustRoot -k '%s' '%s'`, login, cert))
+	addTrustedCertCommand := exec.Command("sh", "-c", fmt.Sprintf(`security add-trusted-cert -k '%s' '%s'`, login, cert))
```

Comment:
⚠️ This is the important change here. The rest is just cleanup and testing. 😎

Reply:
### Testing Results:

#### Test passes without `-d`
```
$ head -n23 dev/ssl_darwin.go | tail -n1 && make test-macos-interactive
	addTrustedCertCommand := exec.Command("sh", "-c", fmt.Sprintf(`security add-trusted-cert -k '%s' '%s'`, login, cert))
This will break your existing puma-dev setup. You'll need to run setup/install again. Cool? Cool.
Also, prepare to provide your system password several times.
Press [return] to continue...
go test ./... -v -test.run=DarwinInteractive -count=1
=== RUN   TestCertificateInstallAndTrustHTTPS_DarwinInteractive
...
--- PASS: TestCertificateInstallAndTrustHTTPS_DarwinInteractive (12.34s)
PASS
...
```

#### Test fails with `-d`
```
$ head -n23 dev/ssl_darwin.go | tail -n1 && make test-macos-interactive
	addTrustedCertCommand := exec.Command("sh", "-c", fmt.Sprintf(`security add-trusted-cert -d -k '%s' '%s'`, login, cert))
This will break your existing puma-dev setup. You'll need to run setup/install again. Cool? Cool.
Also, prepare to provide your system password several times.
Press [return] to continue...
go test ./... -v -test.run=DarwinInteractive -count=1
...
2020/04/29 17:12:34 http: TLS handshake error from 127.0.0.1:57096: remote error: tls: bad certificate
--- FAIL: TestCertificateInstallAndTrustHTTPS_DarwinInteractive (6.13s)
    main_test.go:149:
        	Error Trace:	main_test.go:149
        	            				main_darwin_test.go:115
        	Error:      	Get https://localhost:55443/: x509: certificate signed by unknown authority
        	Test:       	TestCertificateInstallAndTrustHTTPS_DarwinInteractive
FAIL
```